### ISSUE TITLE
`int.__round__(None)` is not supported

### ISSUE THREAD

**Issue by JelleZijlstra** [python/cpython#120080]
# Bug report

### Bug description:

The `__round__` method on floats accepts None as an explicit argument, but the one on int does not:

```python
>>> (1.0).__round__(None)
1
>>> (1).__round__(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object cannot be interpreted as an integer
```

This goes against the general principle that any operation that works on a float should also work on an int.

(Note that both `round(x)` and `round(x, None)` work on both floats and ints; this issue is entirely about direct calls to `.__round__()`.)

### CPython versions tested on:

3.12, CPython main branch

### Linked PRs
* gh-120088
* gh-120197
* gh-120328

**Comment by serhiy-storchaka** [comment]
I do not think that there is a bug. The operation that works on a float and on an int is `round(x, None)`, the rest is an implementation detail. You do not expect that `int.__add__` works on a float, aren't?

**Comment by vstinner** [comment]
Fixed by commit: https://github.com/python/cpython/commit/57ad769076201c858a768d81047f6ea44925a33b

**Comment by serhiy-storchaka** [comment]
Note that `Decimal.__round__` does not accept None. And perhaps many third-party numeric types do not accept it. Implementing this for `int.__round__` sends a false signal that this is required. It won't be long before somebody open an issue about `Decimal.__round__`, and then authors of third-party libraries will be under pressure to implement this unneeded feature in their code. This opened a can of worms, and we do not know what long term consequences will be.

Not everything that can be implemented should be implemented, and I believe that this is such a case.

**Comment by vstinner** [comment]
My motivation to merge the PR was to avoid `<unrepresentable>` in the function signature. This marker causes issues. See for example: https://discuss.python.org/t/ac-null-defaults-prevent-correct-signatures-lets-add-inspect-unrepresentable-to-fix-this/30753

**Comment by serhiy-storchaka** [comment]
#73536 is the general solution of this problem.

**Comment by vstinner** [comment]
> https://github.com/python/cpython/issues/73536 is the general solution of this problem.

Ok, but this issue is open since 2017 and is not fixed yet.

IMO accepting None is just a harmless way to fix the issue.

**Comment by serhiy-storchaka** [comment]
Few months ago I finally created a draft (#117671). I hope this will be included in 3.14.
